TicTacToe.sol

```diff
@@ -40,37 +13,39 @@ contract TicTacToe {
         int8[9] board;
         address winner;
         bool isGameFinished;
+        uint8 movesMade;
     }
 
-    Room[] public rooms; // contract.methods.rooms() -> []Room
-
-    int8 private constant EMPTY_MOVE = 3;
-    int8 private constant O_MOVE = 0;
-    int8 private constant X_MOVE = 1;
-
-    int8[9] private emptyBoard = [
-        EMPTY_MOVE,
-        EMPTY_MOVE,
-        EMPTY_MOVE,
-        EMPTY_MOVE,
-        EMPTY_MOVE,
-        EMPTY_MOVE,
-        EMPTY_MOVE,
-        EMPTY_MOVE,
-        EMPTY_MOVE
+    uint8[3][8] private winningCases = [
+        [0, 1, 2],
+        [3, 4, 5],
+        [6, 7, 8],
+        [0, 3, 6],
+        [1, 4, 7],
+        [2, 5, 8],
+        [0, 4, 8],
+        [2, 4, 6]
     ];
 
+    Room[] public rooms; // contract.methods.rooms() -> []Room
+
     event RoomCreation(uint256 roomNumber);
 
+    error RoomDoesntExist();
+    error PlayerXHasntJoined();
+    error NotYourTurn();
+    error SquareIsTaken();
+    error GameIsFinished();
```

Comment:
These custom errors save gas as opposed to doing all the `requires` `makeMove` previously had